pkg/auth/webapp/service2.go

```diff
@@ -538,6 +551,16 @@ func (s *Service2) afterPost(
 func (s *Service2) deriveFinishRedirectURI(session *Session, graph *interaction.Graph) (redirectURI string) {
 	defer func() {
 		if e, ok := graph.GetAuthenticationInfoEntry(); ok {
+			code, err := s.generateSettingsActionGrant(redirectURI, e)
+			if err != nil {
+				panic(err)
+			}
+
+			if code != "" {
+				e = &authenticationinfo.Entry{
+					ID: code,
+				}
```

Comment:
Why do we need this block?

Reply:
Should not try to override entry ID, instead generate settings action response in consent. Removed.